### DISCUSSION TITLE
Events endpoint - Time Saved

### DISCUSSION THREAD

**Discussion by ThibautMarechal** [vercel/turborepo#1309] [Help] [answered]
Hello, 
I'm writting my own implementation of the remote cache server. (https://github.com/ThibautMarechal/turborepo-remote-cache)
I wanted to fully integrate with everything that turborepo can offer 😄 
But I'm wondering how do you calculate the time saved by using the events endpoint ? (like this: https://vercel.com/docs/concepts/limits/usage#time-saved)

For the moment I just sum all the duration from all HIT events, but it does't take into account tasks launched in parrallel.
It feels weird to show that I saved 5 minutes for a build that usually take 30 seconds (because it execute 10 tasks in parrallel).
Dividing everything by 10 is not a good solution because there is not always 10 tasks, and also the user may also use the --concurrency option.
Does the calculation on vercel side suffer from the same "miss calculation" ?

**Comment by rafaeltab** [comment, marked as answer]
In that case, you did actually save 5 minutes since if you were not using turborepo they would not be in parallel, they would be sequential.
This would mean the time it would normally take is the sum of the tasks, however now it was reduced to a few milliseconds.
